tee-worker/litentry/core/assertion-build/src/a8.rs

```diff
@@ -23,38 +23,175 @@ extern crate sgx_tstd as std;
-use lc_data_providers::graphql::{GraphQLClient, VerifiedCredentialsTotalTxs};
-use litentry_primitives::{Assertion, Identity, ParentchainBlockNumber};
+use lc_data_providers::graphql::{
+	GraphQLClient, VerifiedCredentialsNetwork, VerifiedCredentialsTotalTxs,
+};
+use litentry_primitives::{
+	Assertion, AssertionNetworks, EvmNetwork, Identity, ParentchainBlockNumber, SubstrateNetwork,
+	ASSERTION_NETWORKS,
+};
 use log::*;
 use parachain_core_primitives::VCMPError;
-use std::{str::from_utf8, string::ToString, vec, vec::Vec};
+use std::{collections::HashSet, str::from_utf8, string::ToString, vec, vec::Vec};
+
+lazy_static! {
+	pub static ref NETWORK_HASHSET: HashSet<VerifiedCredentialsNetwork> = {
+		let mut m = HashSet::new();
+
+		let litentry = VerifiedCredentialsNetwork::from(SubstrateNetwork::Litentry);
+		let litmus = VerifiedCredentialsNetwork::from(SubstrateNetwork::Litmus);
+		let polkadot = VerifiedCredentialsNetwork::from(SubstrateNetwork::Polkadot);
+		let kusama = VerifiedCredentialsNetwork::from(SubstrateNetwork::Kusama);
+		let khala = VerifiedCredentialsNetwork::from(SubstrateNetwork::Khala);
+		let ethereum = VerifiedCredentialsNetwork::from(EvmNetwork::Ethereum);
+
+		m.insert(litentry);
+		m.insert(litmus);
+		m.insert(polkadot);
+		m.insert(kusama);
+		m.insert(khala);
+		m.insert(ethereum);
+
+		m
+	};
+}
+
+fn assertion_networks_to_vc_networks(
+	networks: &AssertionNetworks,
+) -> HashSet<VerifiedCredentialsNetwork> {
+	let mut set: HashSet<VerifiedCredentialsNetwork> = HashSet::new();
+
+	if networks.is_empty() {
+		NETWORK_HASHSET.clone()
+	} else {
+		for network in networks {
+			let ret = from_utf8(network.as_ref());
+			match ret {
+				Ok(network) => {
+					let mut network = network.to_string();
+					network.make_ascii_lowercase();
+					let network = network.as_str();
+					if ASSERTION_NETWORKS.contains(&network) {
+						debug!("	[AssertionBuild-A8] available networks: {}", network);
+
+						match network {
+							"litentry" => {
+								let litentry =
+									VerifiedCredentialsNetwork::from(SubstrateNetwork::Litentry);
+								set.insert(litentry);
+							},
+							"litmus" => {
+								let litmus =
+									VerifiedCredentialsNetwork::from(SubstrateNetwork::Litmus);
+								set.insert(litmus);
+							},
+							"polkadot" => {
+								let polkadot =
+									VerifiedCredentialsNetwork::from(SubstrateNetwork::Polkadot);
+								set.insert(polkadot);
+							},
+							"kusama" => {
+								let kusama =
+									VerifiedCredentialsNetwork::from(SubstrateNetwork::Kusama);
+								set.insert(kusama);
+							},
+							"khala" => {
+								let khala =
+									VerifiedCredentialsNetwork::from(SubstrateNetwork::Khala);
+								set.insert(khala);
+							},
+							"ethereum" => {
+								let ethereum =
+									VerifiedCredentialsNetwork::from(EvmNetwork::Ethereum);
+								set.insert(ethereum);
+							},
+							_ => {
+								info!("		[AssertionBuild-A8] Wrong Network!");
+							},
```

Comment:
as `SubstrateNetwork` already derive `Serialize/Deserialize`, maybe we can use [serde_plain](https://crates.io/crates/serde_plain) to simplify this code:

```
impl FromStr for SubstrateNetwork {
    fn from_str() -> Result<String, xxx> {
        // use serde_plain here
    } 
}
```